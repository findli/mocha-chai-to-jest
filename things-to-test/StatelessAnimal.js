import React from 'react';

const propTypes = {
    sound: React.PropTypes.string.isRequired,
};

const StatelessAnimal = ({ sound }) => (
    <div className="animal">
        <p>
            {sound}
        </p>
    </div>
);

StatelessAnimal.propTypes = propTypes;
StatelessAnimal.displayName = 'StatelessAnimal';

export default StatelessAnimal;
