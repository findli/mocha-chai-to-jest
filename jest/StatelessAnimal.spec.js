import React from 'react';
import StatelessAnimal from '../things-to-test/StatelessAnimal';
import ReactTestUtils from 'react-addons-test-utils';

describe('StatelessAnimal component', () => {
    it('renders correctly with min params', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <StatelessAnimal
                sound={"Meow!"}
            />
        );
        expect(result).toMatchSnapshot();
    });
});
