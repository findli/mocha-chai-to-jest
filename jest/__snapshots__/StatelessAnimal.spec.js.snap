exports[`StatelessAnimal component renders correctly with min params 1`] = `
<div
  className="animal">
  <p>
    Meow!
  </p>
</div>
`;
